lib/AST/ASTPrinter.cpp

```diff
@@ -1472,6 +1472,10 @@ static void reconstituteInverses(GenericSignature genericSig,
   for (auto tp : typeParams) {
     assert(tp);
 
+    // Any generic parameter requiring a class could not have an inverse.
+    if (genericSig->requiresClass(tp))
```

Comment:
does `<T where T: AnyObject, T: Copyable>` minimize as `<T where T: AnyObject>`? It should via a concrete conformance but I believe that needs to be implemented in the requirement machine itself.

Reply:
Maybe instead of doing `sig->requiresProtocol(paramTy, Copyable)` you can map paramTy into the signature's environment and do a global lookup with the archetype to Copyable? Then you won't need this special case.